src/library_syscall.js

```diff
@@ -1815,17 +1823,27 @@ if (SYSCALL_DEBUG) {
   }
 }
 
+var WASI_SYSCALLS = set(['fd_write']);
+
 for (var x in SyscallsLibrary) {
+  var which = null; // if this is a musl syscall, its number
   var m = /^__syscall(\d+)$/.exec(x);
-  if (!m) continue;
-  var which = +m[1];
+  if (m) {
+    which = +m[1];
+  } else {
+    if (!(x in WASI_SYSCALLS)) {
+      continue;
+    }
+  }
   var t = SyscallsLibrary[x];
   if (typeof t === 'string') continue;
   t = t.toString();
   var pre = '', post = '';
-  pre += 'SYSCALLS.varargs = varargs;\n';
+  if (which) {
+    pre += 'SYSCALLS.varargs = varargs;\n';
+  }
 #if SYSCALL_DEBUG
-  pre += "err('syscall! ' + [" + which + ", '" + SYSCALL_CODE_TO_NAME[which] + "']);\n";
+  pre += "err('syscall! ' + [" + which + ", '" + (which ? SYSCALL_CODE_TO_NAME[which] : x) + "']);\n";
```

Comment:
Presumably we don't want to use `which` at all here since it could be null?

How about two different variables:

```
var syscall_number = null;
var syscall_name = x;
if (m) {
  syscall_number = +m[1];
  syscall_name = SYSCALL_CODE_TO_NAME[syscall_number] + '(' + x + ')';
}
```

Reply:
Good idea. I think I can simplify this even further, as there is no separate number/name for a non-musl syscall. Pushing a fix now.